src/nrncvode/netcvode.cpp

```diff
@@ -6085,7 +6085,7 @@ void VecRecordDiscrete::frecord_init(TQItem* q) {
 
 void VecRecordDiscrete::deliver(double tt, NetCvode* nc) {
     y_->push_back(*pd_);
-	assert(Math::equal(y_->vec().back(), tt, 1e-8));
+	assert(Math::equal(t_->vec().back(), tt, 1e-8));
```

Comment:
Sorry about that, It was so obvious I didn't bother to test. The correct line is actually (tested)
```
--- a/src/nrncvode/netcvode.cpp
+++ b/src/nrncvode/netcvode.cpp
@@ -6085,9 +6085,9 @@ void VecRecordDiscrete::frecord_init(TQItem* q) {
 
 void VecRecordDiscrete::deliver(double tt, NetCvode* nc) {
     y_->push_back(*pd_);
-       assert(Math::equal(y_->vec().back(), tt, 1e-8));
+       assert(Math::equal(t_->vec()[y_->size()-1], tt, 1e-8));
        if (y_->size() < t_->size()) {
-               e_->send(t_->vec().back(), nc, nrn_threads);
+               e_->send(t_->vec()[y_->size()], nc, nrn_threads);
        }
 }
```
Or you may prefer storing  y_.size()  in a temporary and using that in the two place.